### PR TITLE
Homepage: Add initial carousel organism

### PR DESCRIPTION
**Pull request by anselmbradford**
Adds an initial front-end architecture for a carousel organism, for cycling through a sequence of blocks of content.

## Additions

- Adds Carousel organism JavaScript, CSS, and jinja template.
- Adds `single.js` script for scripts only on the homepage route.

## Testing

1. Pull branch, run `gulp clean && gulp build`
2. Enable alternate homepage design (via instructions in https://github.com/cfpb/cfgov-refresh/pull/5335)
3. Visit http://localhost:8000/?nhp=True and see there's a barebones carousel that you can click forward and backward through.

## Screenshots

carousel

### CODE REVIEW

**Comment by anselmbradford** [comment]
@contolini

File: cfgov/unprocessed/css/organisms/carousel.less [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  // Override default unordered list padding.
+  & ul {
+    padding: 0;
+  }
+
+  &_item {
+    border: 1px solid red;
+    height: 100px;
+    width: 100%;
```
Comment (contolini) [inline cfgov/unprocessed/css/organisms/carousel.less]: Down the road, how do you envision image aspect ratios being handled? Would we maintain a fixed height, center the image in the container, and overflow the sides? i.e.

0NrWT

Comment (contolini) [reply cfgov/unprocessed/css/organisms/carousel.less]: In the same vein, we'll have to reuse these images for the previews by somehow cropping them into little squares. We could use the same technique as above. Loading the full-size images in the previews would preload them for us.

Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/carousel.less]: That's a good question. One direction might be to follow the approach used in https://cfpb.github.io/design-system/components/featured-content-module

**Comment by anselmbradford** [comment]
> This works well on the English homepage, but doesn't work properly on the Spanish one -- the panels appear, but the JS isn't loaded. Is that because the single.js file isn't loaded against that Spanish URL?

Ah, yes, that's correct. How do you think we should handle that? We can create a duplicate `single.js` in `/es/`, which is a shame, but I was also thinking this would be in Wagtail at some point and Carousel.js would move to the `on-demand` directory.

**Comment by chosak** [comment]
@anselmbradford per our conversation offline, I just pushed a commit that moves the homepage JS to a new `routes/on-demand/home-page.js` file, referenced by our [custom, but infrequently used](https://github.com/cfpb/cfgov-refresh/search?l=Python&q=page_js&type=) [`page_js`](https://github.com/cfpb/cfgov-refresh/blob/5badef39b1e49269f1b37058f3d6478d7a74baf4/cfgov/v1/models/base.py#L297) property.

**Pull request merged**